emr-user-role-mapper-application/src/main/java/com/amazon/aws/emr/credentials/STSCredentialsProvider.java

```diff
@@ -49,10 +52,27 @@
         });
 
     synchronized static AWSSecurityTokenService getStsClient() {
+        try {
+            if (region == null) {
+                region = Regions.getCurrentRegion();
+            }
+        } catch (Exception e){
+            log.info("Could not determine the AWS region!");
+        }
         if (stsClient == null) {
-            stsClient = AWSSecurityTokenServiceClientBuilder
+            if (region != null) {
+                String endpoint = String.format("https://sts.%s.amazonaws.com", region.getName());
```

Comment:
it's better to make the regional endpoint url inside a config which would allow users to override/update the pattern if something different come out in future. 

In the config, you will have a global pattern which will work if no regional override exists. If there's special region which follows a different format, we should allow users to explicitly add that as an override.